packages/matter.js/src/cluster/ClusterServerFactory.ts

```diff
@@ -0,0 +1,122 @@
+
+    // TODO - caller can override but should we make the set of optional
+    // attributes configurable?
+    optionalAttributeNames(cluster.attributes).forEach((name) => {
+        if (cluster.attributes[name] !== undefined) {
+            (result as any)[name] = cluster.attributes[name].default
+        }
+    });
+
+    return result;
+}
+
+/**
+ * This factory acts as a central registry for default cluster implementations.
+ */
+export class ClusterServerFactory {
+    private static defaultHandlerProviders = new Map<AnyCluster, () => any>();
+    private static defaultAttributeInitialValues = new Map<AnyCluster, any>;
+
+    /**
+     * Retrieve default values for attributes based on cluster schema.
+     */
+    public static getDefaultAttributeInitialValues<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(cluster: Cluster<F, SF, A, C, E>) {
+        const aiv = this.defaultAttributeInitialValues.get(cluster) as AttributeInitialValues<A>;
+        if (aiv == undefined) {
+            return createDefaultAttributeInitialValues(cluster);
+        }
+        return aiv;
+    }
+
+    /**
+     * Register the default implementation for a cluster.
+     * 
+     * @param cluster the cluster to configure
+     * @param defaultHandlerProvider a function that create the cluster handlers
+     * @param defaultAttributeInitialValues optional overrides for default attribute values
+     */
+    public static registerClusterDefaults<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(
+        cluster: Cluster<F, SF, A, C, E>, defaultHandlerProvider: () => ClusterServerHandlers<Cluster<F, SF, A, C, E>>,
+        defaultAttributeInitialValues: AttributeInitialValues<A> = createDefaultAttributeInitialValues(cluster)
+    ) {
+        this.defaultHandlerProviders.set(cluster, defaultHandlerProvider);
+        this.defaultAttributeInitialValues.set(cluster, defaultAttributeInitialValues);
+
+        const x = this.defaultAttributeInitialValues.get(cluster);
+        if (x.sceneCount) console.log(x.sceneCount);
+    }
+
+    /**
+     * Create a new cluster server.
+     * 
+     * @param cluster the cluster for the server
+     * @param attributeInitialValues optional overrides for attribute values
+     * @param handlers optional overrides for cluster handlers
+     * @returns 
+     */
+    public static create<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(cluster: Cluster<F, SF, A, C, E>, attributeInitialValues?: AttributeInitialValues<A>, handlers?: ClusterServerHandlers<Cluster<F, SF, A, C, E>>) {
+        handlers = Object.assign({}, handlers);
+
+        const provider = this.defaultHandlerProviders.get(cluster);
+        if (provider) {
+            Object.assign(handlers, provider());
+        }
+
+        for (const command in cluster.commands) {
```

Comment:
Formally here the same with optional commands. We should not define them. In General er need a way later to identify implemented/supported commands vs not implemented ones - or need to populate global attributes here already.

Reply:
Sorry, forgot to take that out.